src/arch/riscv/faults.hh

```diff
@@ -93,6 +94,54 @@ enum ExceptionCode : uint64_t
     INT_EXT_USER = 8,
     INT_EXT_SUPER = 9,
     INT_EXT_MACHINE = 11,
+    INT_LOCAL_16 = 16,
```

Comment:
I would like to start with 0 like `INT_LOCAL_0, INT_LOCAL_1 ...`

Reply:
Do you mean `INT_LOCAL_0` to `INT_LOCAL_47`?

I do not understand your reference. In https://five-embeddev.com/quickref/interrupts.html, the local interrupts also start with 16 (`platform_defined16, platform_defined17 ...`)

Reply:
> Do you mean `INT_LOCAL_0` to `INT_LOCAL_47`?

Yes

> I do not understand your reference. In https://five-embeddev.com/quickref/interrupts.html, the local interrupts also start with 16 (`platform_defined16, platform_defined17 ...`)

It means the interrupt ID 16 - X is defined by platform. It comment `Optional platform defined interrupt source 0.` in the description field for the interrupt ID 16. So I think use `INT_LOCAL_0` will make it clear to know the interrupt source it from 0.

Another example for using local interrupt in SiFive E31:

 
![clint interrupt](https://github.com/gem5/gem5/assets/32214817/43a9b839-f335-42e5-91c7-49dd8024cfcd)

Reply:
I forgot to refresh my browser before writing the comment above. Thank you for the examples. I will change this.